R/utilities-task.R

```diff
@@ -61,14 +61,18 @@ inactivateWorkflowTasks <- function(workflow, tasks = workflow$getAllTasks()) {
 loadSimulateTask <- function(workflow, active = TRUE, settings = NULL) {
   validateIsOfType(workflow, "Workflow")
   validateIsLogical(active)
-  taskFunction <- simulateModel
-  nameFunction <- deparse(substitute(simulateModel))
+
+  taskFunction <- simulateModelParallel
+  nameFunction <- deparse(substitute(simulateModelParallel)) # deparse(substitute(simulateModel))
+  simulationTaskInitializer <- ParallelSimulationTask$new
+
   if (isOfType(workflow, "PopulationWorkflow")) {

```

Comment:
Where do you check that there is not a mix of Population and Individual Simulations?
Also what about putting everything in one task? Why is the new task required

Reply:
I am assuming that the simulation sets are either for individuals (mean model) or for populations.  

It is possible to implement some logic in the `runTask` function of the `SimulationTask` object and remove the `ParallelSimulationTask`, but I felt this way would be neater (even if it means an extra class).  

The `runTask` function receives a list of `structureSet` objects.  This list doesn't tell us immediately whether `structureSets` comes from a mean workflow or a population workflow.  We would need to inspect the `structureSet` elements individually to see if any one of them stores a path to a population file, and if so, simulate each `structureSet` simulation in sequence using the `simulateModelForPopulation` as the `taskFunction` (otherwise simulate all the mean model simulations in parallel using the `simulateModelParallel` task function).

Instead, here, the appropriate `taskFunction` is selected based on the workflow.  If it is a mean model workflow, use a `ParallelSimulationTask` which passes the `structureSets` list to `simulateModelParallel` (the `taskFunction`) for parallel simulation.  If it is a population workflow, then loop thru the `structureSets` list and run each simulation separately using the `simulateModelForPopulation` as the `taskFunction`.

Reply:
>I am assuming that the simulation sets are either for individuals (mean model) or for populations.

Is this the correct assumptions? Can't we have a workflow where things are mixed? Asking for a friend

>but I felt this way would be neater (even if it means an extra class).
I don't mind the extra class. Let's keep it

also you can use `if (isOfType(workflow, PopulationWorkflow))`  (e.g. class instead of string)

The only problem I can see if that if we add more Workflows, we need to update this code. So we lose some flexibility. But good for now

How many workflows do we have?

Reply:
Done.  

We have: `MeanModelWorkflow` `PopulationWorkflow` `QualificationWorkflow`

So far workflows aren't mixed, they are either individual (mean/qualification) or population.  I'm guessing this may indeed change for qualification (@Yuri05 ?)